src/main/java/tc/oc/pgm/spawns/states/Dead.java

```diff
@@ -79,6 +81,11 @@ public void tick() {
       player.resetGamemode();
     }
 
+    if (player.getSettings().getValue(SettingKey.RESPAWN) == SettingValue.RESPAWN_AUTO
+        && ticksUntilRespawn() <= 0) {
+      super.requestSpawn();
+    }
+
```

Comment:
I don't think there's much of a benefit to calculating this each tick. Just set `spawnRequested` to true if this setting matches on the constructor of `Dead`